### DISCUSSION TITLE
Feature Request: Unarchive option and option to display archive size.

### DISCUSSION THREAD

**Discussion by r7l** [go-shiori/shiori#770] [Feature requests]
This is related to this issue: https://github.com/go-shiori/shiori/issues/474 

I've set my instance to always create an archive. This is an awesome feature! But some sites end up with ridiculous huge archives.

This site is a good example: https://skylum.com/luminar 

When bookmarking it with archive enabled, it creates a **175mb** archive file. The worst part of it is, that the archive content is pretty much useless. It's barely showing anything. It would really help to have an option to unarchive a site later on. Currently, i can only see an option to redo the archive, which does not help. So, the only way to remove the archive would be by deleting and redoing the bookmark.

The second part of my feature request would be an option to show the size of the archive somewhere on the bookmark box. For example right next to the ID (or instead if deactived). This way it would be easier to spot huge archives and decide right away. Currently one has to go through the files manually on the filesystem.

**Comment by fmartingr** [comment]
As a workaround, you can remove your files manually as well using the bookmark ID. @Monirzadeh already had some ideas to display how much size the database/files were taking up, but a new cache option to delete the files from a specific bookmark may be useful as well. We could also denormalize the filesize in database, though I do not think we could modify the search to sort by filesize soon. Maybe this new admin widget could show the top 10 files by size, like some operating systems do. I will give this a thought and try to create some issues from it.
